### ISSUE TITLE
Visual Studio 2019 support

### ISSUE THREAD

**Issue by smanders** [smanders/externpro#307]
Visual Studio 2017
* https://github.com/smanders/externpro/issues/185
* https://github.com/smanders/externpro/issues/189

**Comment by smanders** [comment]
nodejs failure
```
Performing configure step for 'nodejs_10.17.0_Releasevcbuild'
Note: vcbuild no longer signs by default. "nosign" is redundant.
Looking for Python 2.x
Looking for NASM
externpro C:\Users\smanders\src\externpro\_bld\xpbase\Source\nodejs_10.17.0\tools\msvs\..\..\..\nasm
Looking for Visual Studio 2017
Failed to find a suitable Visual Studio installation.
Try to run in a "Developer Command Prompt" or consult
https://github.com/nodejs/node/blob/master/BUILDING.md#windows-1
```

**Comment by smanders** [comment]
```
Performing build_Release_x64 step for 'llvmformat_msvc'
Microsoft (R) Build Engine version 16.8.2+25e4d540b for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  obj.clang-tblgen.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\tools\clang\utils\TableGen\obj.clang-tblgen.dir\Release\obj.clang-tblgen.lib
  LLVMSupport.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\lib\LLVMSupport.lib
  LLVMTableGen.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\lib\LLVMTableGen.lib
  clang-tblgen.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\bin\clang-tblgen.exe
  AsmWriterEmitter.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\xtree(1705,1): error C3848: expression having type 'const `anonymous-namespace'::AliasPriorityComparator' would lose some const-volatile qualifiers in order to call 'bool `anonymous-namespace'::AliasPriorityComparator::operator ()(const `anonymous-namespace'::AliasPriorityComparator::ValueType &,const `anonymous-namespace'::AliasPriorityComparator::ValueType &)' [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\xtree(1009): message : see reference to function template instantiation 'std::_Tree_find_result<std::_Tree_node<std::pair<llvm::CodeGenInstAlias,int>,std::_Default_allocator_traits<_Alloc>::void_pointer> *> std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::_Find_lower_bound<_Key>(const _Keyty &) const' being compiled [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
          with
          [
              _Alloc=std::allocator<std::pair<llvm::CodeGenInstAlias,int>>,
              _Kty=std::pair<llvm::CodeGenInstAlias,int>,
              _Pr=`anonymous-namespace'::AliasPriorityComparator,
              _Key=std::pair<llvm::CodeGenInstAlias,int>,
              _Keyty=std::pair<llvm::CodeGenInstAlias,int>
          ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\xtree(1009): message : see reference to function template instantiation 'std::_Tree_find_result<std::_Tree_node<std::pair<llvm::CodeGenInstAlias,int>,std::_Default_allocator_traits<_Alloc>::void_pointer> *> std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::_Find_lower_bound<_Key>(const _Keyty &) const' being compiled [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
          with
          [
              _Alloc=std::allocator<std::pair<llvm::CodeGenInstAlias,int>>,
              _Kty=std::pair<llvm::CodeGenInstAlias,int>,
              _Pr=`anonymous-namespace'::AliasPriorityComparator,
              _Key=std::pair<llvm::CodeGenInstAlias,int>,
              _Keyty=std::pair<llvm::CodeGenInstAlias,int>
          ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\xtree(1344): message : see reference to function template instantiation 'std::pair<std::_Tree_node<std::pair<llvm::CodeGenInstAlias,int>,std::_Default_allocator_traits<_Alloc>::void_pointer> *,bool> std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::_Emplace<std::pair<llvm::CodeGenInstAlias,int>>(std::pair<llvm::CodeGenInstAlias,int> &&)' being compiled [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
          with
          [
              _Alloc=std::allocator<std::pair<llvm::CodeGenInstAlias,int>>,
              _Kty=std::pair<llvm::CodeGenInstAlias,int>,
              _Pr=`anonymous-namespace'::AliasPriorityComparator
          ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\xtree(1344): message : see reference to function template instantiation 'std::pair<std::_Tree_node<std::pair<llvm::CodeGenInstAlias,int>,std::_Default_allocator_traits<_Alloc>::void_pointer> *,bool> std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::_Emplace<std::pair<llvm::CodeGenInstAlias,int>>(std::pair<llvm::CodeGenInstAlias,int> &&)' being compiled [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
          with
          [
              _Alloc=std::allocator<std::pair<llvm::CodeGenInstAlias,int>>,
              _Kty=std::pair<llvm::CodeGenInstAlias,int>,
              _Pr=`anonymous-namespace'::AliasPriorityComparator
          ]
C:\Users\smanders\src\externpro\_bld\xpbase\Source\llvm\utils\TableGen\AsmWriterEmitter.cpp(774): message : see reference to function template instantiation 'std::pair<std::_Tree_const_iterator<std::_Tree_val<std::_Tree_simple_types<std::pair<llvm::CodeGenInstAlias,int>>>>,bool> std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::insert<false,0>(std::pair<llvm::CodeGenInstAlias,int> &&)' being compiled [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
          with
          [
              _Kty=std::pair<llvm::CodeGenInstAlias,int>,
              _Pr=`anonymous-namespace'::AliasPriorityComparator,
              _Alloc=std::allocator<std::pair<llvm::CodeGenInstAlias,int>>
          ]
C:\Users\smanders\src\externpro\_bld\xpbase\Source\llvm\utils\TableGen\AsmWriterEmitter.cpp(773): message : see reference to function template instantiation 'std::pair<std::_Tree_const_iterator<std::_Tree_val<std::_Tree_simple_types<std::pair<llvm::CodeGenInstAlias,int>>>>,bool> std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::insert<false,0>(std::pair<llvm::CodeGenInstAlias,int> &&)' being compiled [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
          with
          [
              _Kty=std::pair<llvm::CodeGenInstAlias,int>,
              _Pr=`anonymous-namespace'::AliasPriorityComparator,
              _Alloc=std::allocator<std::pair<llvm::CodeGenInstAlias,int>>
          ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\xtree(1695,1): error C3848: expression having type 'const `anonymous-namespace'::AliasPriorityComparator' would lose some const-volatile qualifiers in order to call 'bool `anonymous-namespace'::AliasPriorityComparator::operator ()(const `anonymous-namespace'::AliasPriorityComparator::ValueType &,const `anonymous-namespace'::AliasPriorityComparator::ValueType &)' [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\xtree(1010): message : see reference to function template instantiation 'bool std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::_Lower_bound_duplicate<_Key>(std::_Tree_node<std::pair<llvm::CodeGenInstAlias,int>,std::_Default_allocator_traits<_Alloc>::void_pointer> *const ,const _Keyty &) const' being compiled [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
          with
          [
              _Kty=std::pair<llvm::CodeGenInstAlias,int>,
              _Pr=`anonymous-namespace'::AliasPriorityComparator,
              _Alloc=std::allocator<std::pair<llvm::CodeGenInstAlias,int>>,
              _Key=std::pair<llvm::CodeGenInstAlias,int>,
              _Keyty=std::pair<llvm::CodeGenInstAlias,int>
          ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\xtree(1010): message : see reference to function template instantiation 'bool std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::_Lower_bound_duplicate<_Key>(std::_Tree_node<std::pair<llvm::CodeGenInstAlias,int>,std::_Default_allocator_traits<_Alloc>::void_pointer> *const ,const _Keyty &) const' being compiled [C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\utils\TableGen\obj.llvm-tblgen.vcxproj]
          with
          [
              _Kty=std::pair<llvm::CodeGenInstAlias,int>,
              _Pr=`anonymous-namespace'::AliasPriorityComparator,
              _Alloc=std::allocator<std::pair<llvm::CodeGenInstAlias,int>>,
              _Key=std::pair<llvm::CodeGenInstAlias,int>,
              _Keyty=std::pair<llvm::CodeGenInstAlias,int>
          ]
  LLVMMC.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\lib\LLVMMC.lib
  clangAST.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\lib\clangAST.lib
  clangBasic.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\lib\clangBasic.lib
  clangFormat.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\lib\clangFormat.lib
  clangLex.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\lib\clangLex.lib
  clangRewrite.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\lib\clangRewrite.lib
  clangToolingCore.vcxproj -> C:\Users\smanders\src\externpro\_bld\xpbase\Build\llvmformat_msvc\Release\lib\clangToolingCore.lib
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(238,5): error MSB8066: Custom build for 'C:\Users\smanders\src\externpro\_bld\CMakeFiles\e9de4786f30f071beee1a5fac7ae9c99\llvmformat_msvc-build_Release_x64.rule;C:\Users\smanders\src\externpro\_bld\CMakeFiles\e9de4786f30f071beee1a5fac7ae9c99\llvmformat_msvc-install.rule;C:\Users\smanders\src\externpro\_bld\CMakeFiles\e9de4786f30f071beee1a5fac7ae9c99\llvmformat_msvc-install_Release_x64.rule;C:\Users\smanders\src\externpro\_bld\CMakeFiles\a9068d26fb35d35db2a45e834a1a0484\llvmformat_msvc-complete.rule;C:\Users\smanders\src\externpro\_bld\CMakeFiles\2b710e3b9fa1ac2158f3e6fc0d715fab\llvmformat_msvc.rule' exited with code 1.
Done building project "llvmformat_msvc.vcxproj" -- FAILED.
```

**Comment by smanders** [comment]
boost 1.67 fails, so wxinclude and fecpp (which depend on boost) also fail

**Comment by smanders** [comment]
these issues were addressed in the 21.05 milestone https://github.com/smanders/externpro/milestone/32?closed=1
* nodejs https://github.com/smanders/externpro/issues/272
* llvm/clang/clang-format https://github.com/smanders/externpro/issues/312
* boost https://github.com/smanders/externpro/issues/308
